src/opossum_lib/opossum_file.py

```diff
@@ -60,35 +61,72 @@ class Metadata:
     buildDate: Optional[str] = None
 
 
+class ResourceType(Enum):
+    FILE = auto()
+    FOLDER = auto()
+    TOP_LEVEL = auto()
+    OTHER = auto()
+
+
 @dataclass(frozen=True)
 class Resource:
+    type: ResourceType
     children: Dict[str, Resource] = field(default_factory=dict)
 
-    def add_path(self, path: List[str]) -> None:
-        if len(path) == 0:
+    def add_path(
+        self, path_with_resource_types: List[Tuple[str, ResourceType]]
+    ) -> None:
+        if len(path_with_resource_types) == 0:
             return
-        first, rest = path[0], path[1:]
+        (first, resource_type), rest = (
+            path_with_resource_types[0],
+            path_with_resource_types[1:],
+        )
+        if self.element_exists_but_resource_type_differs(first, resource_type):
+            raise TypeError(
+                "Couldn't add path to resource: ResourceType of elements with"
+                " the same path differ."
+            )
         if first not in self.children:
-            self.children[first] = Resource()
+            self.children[first] = Resource(type=resource_type)
         self.children[first].add_path(rest)
 
+    def element_exists_but_resource_type_differs(
+        self, element: str, type_of_last_element: ResourceType
```

Comment:
rename `type_of_last_element` to something like `resource_type` or `expected_resource_type`